twill-yarn/src/main/java/org/apache/twill/filesystem/FileContextLocation.java

```diff
@@ -162,7 +163,8 @@ public URI toURI() {
     // append "port" to the path URI, while the DistributedFileSystem always use the cluster logical
     // name, which doesn't allow having port in it.
     URI uri = path.toUri();
-    if (HAUtil.isLogicalUri(locationFactory.getConfiguration(), uri)) {
+
+    if (FileContextLocationUtil.useLogicalUri(locationFactory.getConfiguration(), uri)) {
```

Comment:
It seems like we actually don't need to check, but rather just always strip off the port from the `URI`, since the `FileContext` or `FileSystem` in Hadoop will always determine it internally based on the `Configuration`.

In fact that would make the returning `URI` more predictable.